src/main/java/de/tum/in/www1/artemis/service/connectors/athena/AthenaConnector.java

```diff
@@ -0,0 +1,104 @@
+package de.tum.in.www1.artemis.service.connectors.athena;
+
+import javax.validation.constraints.NotNull;
+
+import org.slf4j.Logger;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.client.ResourceAccessException;
+import org.springframework.web.client.RestTemplate;
+
+import de.tum.in.www1.artemis.exception.NetworkingException;
+
+/**
+ * Connector to Athena, a remote Artemis service that can create semi-automatic feedback suggestions for tutors.
+ * <p>
+ * The connector will be used to
+ * - Send submissions to Athena
+ * - Send existing feedback to Athena
+ * - Request the next suggested submission to assess from Athena
+ * - Receive feedback suggestions from Athena
+ * <p>
+ * As parameters, this class required DTO classes to serialize and deserialize POJOs to JSON and back.
+ *
+ * @param <RequestType>  DTO class, describing the body of the network request.
+ * @param <ResponseType> DTO class, describing the body of the network response.
+ */
+class AthenaConnector<RequestType, ResponseType> {
+
+    private final Logger log;
+
+    private final RestTemplate restTemplate;
+
+    private final Class<ResponseType> genericResponseType;
+
+    AthenaConnector(Logger log, RestTemplate restTemplate, Class<ResponseType> genericResponseType) {
+        this.log = log;
+        this.restTemplate = restTemplate;
+        this.genericResponseType = genericResponseType;
+    }
+
+    /**
+     * Invoke Athena with a network call.
+     *
+     * @param url           Athena api endpoint
+     * @param requestObject request body as POJO
+     * @return response body from Athena
+     * @throws NetworkingException exception in case of unsuccessful responses or responses without a body.
+     */
+    private ResponseType invoke(@NotNull String url, @NotNull RequestType requestObject) throws NetworkingException {
+        long start = System.currentTimeMillis();
+        log.debug("Calling Athena.");
+
+        var headers = new HttpHeaders();
+        headers.setContentType(MediaType.APPLICATION_JSON);
+
+        final HttpEntity<RequestType> httpRequestEntity = new HttpEntity<>(requestObject, headers);
+
+        ResponseEntity<ResponseType> response;
+        try {
+            response = restTemplate.postForEntity(url, httpRequestEntity, genericResponseType);
+        }
+        catch (ResourceAccessException e) {
+            log.error("Athena did not respond successfully in time: {}, stack trace: {}", e.getMessage(), e.getStackTrace());
```

Comment:
You can just add the exception without any placeholders, the whole stack trace gets logged automatically.
```suggestion
            log.error("Athena did not respond successfully in time", e);
```